OsmAnd-java/src/net/osmand/util/GeoPointParserUtil.java

```diff
@@ -813,15 +844,57 @@ public boolean isGeoPoint() {
 			return geoPoint;
 		}
 
+		private String formatDouble(double d) {
+			if(d == (long) d)
+				return String.format(Locale.ENGLISH, "%d", (long)d);
+			else
+				return String.format("%s", d);
+		}
+
 		public boolean isGeoAddress() {
 			return geoAddress;
 		}
 
+		/**
+		 * Generates a URI string according to https://tools.ietf.org/html/rfc5870 and
+		 * https://developer.android.com/guide/components/intents-common.html#Maps
+		 */
 		@Override
 		public String toString() {
```

Comment:
URI should not be generated in toString() method, there should be a separate method  for this. toString should be used only for debugging feature preferrably